src/functions/functions.ts

```diff
@@ -27,6 +27,14 @@ export function clock(invocation: CustomFunctions.StreamingInvocation<string>):
   };
 }
 
+/**
+ * Returns the current time.
+ * @returns String with the current time formatted for the current locale.
+ */
+export function currentTime(): string {
+  return new Date().toLocaleTimeString();
```

Comment:
Why does putting a new function make it work?

Reply:
That's not what made it work . . . that was just to be more uniform with the base CF template.  The manifest changes (adding dist to the html path) are what correct this.  The other changes to the manifest point the functions page to the taskpane page instead of the funtions.html page and CF needs the pages from the dist folder.

Reply:
For the shared runtime, you _shouldn't_ need the dist folder, and it's preferable to not have it so you don't have to do a separate "npm run build" before "npm start" (or start debugging in VS code).

Reply:
Well . . . it looks like that didn't quit solve the problem anyway.

We are getting inconsistent results between our local machines and the clean test machines (which unfortunately I can't get checked out on failure) and we've been ignoring it in previous merges.  Test pass locally, but not in the pipeline.  I intend to fix that this time.  I've been using a different branch to play with job submissions that have incremental changes between master and what the shared-runtime branch looks like.  The change that triggers the test start failing is:

- changing '<SourceLocation resid="Functions.Page.Url"/>' to '<SourceLocation resid="Taskpane.Url"/>'.  

The two resources are 

- '<bt:Url id="Functions.Page.Url" DefaultValue="https://localhost:3001/dist/functions.html"/>'
- '<bt:Url id="Taskpane.Url" DefaultValue="https://localhost:3001/taskpane.html"/>'

We can see two differences in the URL:

- "functions.html" vs. "taskpane.html"
- dist folder vs. no dist folder

The change to use taskpane.html is part of the instructions used to convert the output of the yo-office branch to work in shared runtime and is intentional.  In my test branch changing Functions.Page.Url to use taskpane.html caused allowed the test to pass and trying to directly browse the two locations ("https://localhost:3001/taskpane.html" and "https://localhost:3001/dist/taskpane.html" ) on a local running dev server show that the root location has errors when trying to render and the dist location does not.  So I assumed that using "dist" in the Taskpane.Url would work and made that change here . . . but the test still fails.

So I'm back to investigating.  Any suggestions on what to look for would be appreciated.

Reply:
After further investigation it looks like the failure occurs when the Runtime reference and the page source location reference are pointing to the same resource id.  If I make two different resource ids pointing to the same URL and then set each to a different one then the test passes.